brooklyn-server/core/src/test/java/org/apache/brooklyn/core/config/ConfigKeyConstraintTest.java

```diff
@@ -297,12 +299,17 @@ public void testContextAwarePredicateInformedOfEntity() {
     public void testQuickFutureResolved() {
         // Result of task is -1, outside of the range specified by the config key.
         try {
-            app.createAndManageChild(EntitySpec.create(EntityRequiringConfigKeyInRange.class)
+            EntityRequiringConfigKeyInRange child = app.createAndManageChild(EntitySpec.create(EntityRequiringConfigKeyInRange.class)
                     .configure(EntityRequiringConfigKeyInRange.RANGE, sleepingTask(Duration.ZERO, -1)));
-            fail("Expected exception when managing entity with incorrect config");
+            // may or may not fail above, depending on speed, but should fail if assert after forcing resolution
```

Comment:
I think the second option would be too surprising. It may not be the caller's fault if the value is invalid but it's always the case when setting values.